src/server/game/Entities/Creature/GossipDef.h

```diff
@@ -237,6 +234,21 @@ class TC_GAME_API QuestMenu
         QuestMenuItemList _questMenuItems;
 };
 
+class SelectionData
+{
+    public:
+        SelectionData() { Reset(); }
+
+        void Reset()
+        {
+            SenderGuid = ObjectGuid::Empty;
+            TrainerId = 0;
+        }
+
+        ObjectGuid SenderGuid;
```

Comment:
Interaction sure, but TargetGuid I am not sure about.
I would rather use Sender or Source or something like that myself. I also think in may places source is used in core already.